src/dashboard/queue_distribute/queue_distribute.py

```diff
@@ -0,0 +1,102 @@
+"""Handler for submitting studies to an SNS queue for distribution to remote sites.
+
+In the long term, this module (or a submodule imported by this module) will be responsible
+for parsing the output of a dashboard guided workflow/query builder generated payload,
+and converting it to a cumulus library compatible study.
+
+It also enables distribution of a smart-on-fhir owned study directly from github,
+which is the mode it operates in today.
+
+Due to needing cumulus library, this handler is different from all the other lambda
+handlers, in that it is packaged in a docker image and loaded from an elastic container
+registry. This has a few architectural implications not present in other lambdas - notably,
+the home directory is static and we have to write any data to disk inside of /tmp as the
+only writable location.
+
+"""
+
+import json
+import os
+import pathlib
+import subprocess
+
+import boto3
+from cumulus_library import cli
+
+from shared import decorators, functions
+
+# lambda performance tuning - moving these two variables to be global in the module
+# means that their initialization happens during build, rather than invocation.
+# This helps reduce the spinup time, especially for the boto client, and since
+# there are some hefty bits in here already with the docker spinup, shaving a second
+# or two off here is helpful to keep us under the ten second timeout.
+#
+# https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html#function-code
+
+sns_client = boto3.client("sns", os.environ.get("AWS_REGION"))
+BASE_DIR = "/tmp"  # noqa: S108
+
+
+def get_study_from_github(config):
+    try:
+        subprocess.run(["/usr/bin/git", "clone", config["url"], f"{BASE_DIR}/studies"], check=True)  # noqa: S603
+        if "tag" in config and config["tag"] is not None:
+            subprocess.run(  # noqa: S603
+                ["/usr/bin/git", "checkout", "tag"],
```

Comment:
I [think](https://stackoverflow.com/questions/20280726/how-to-clone-a-specific-git-tag) you can pass `--branch <tag>` to the clone command instead of doing a separate checkout (and maybe also `--depth 1` for performance).

Reply:
ok, perf, i'll take it